src/main.cpp

```diff
@@ -749,6 +749,18 @@ int main(int argc, char**argv)
        it != remote_session_stores_locations.end();
        ++it)
   {
+    if (options.memcached_write_format == MemcachedWriteFormat::JSON)
+    {
+      serializer = new SessionStore::JsonSerializerDeserializer();
```

Comment:
It's worth defining this as a new SessionStore::SerializerDeserializer* and std::vector<SessionStore::SerializerDeserializer*> so it's clear these are separate for each remote store (I think this only works atm as they've been set to NULL/cleared in the constructor).